index.ts

```diff
@@ -274,12 +290,34 @@ export default class Analytics {
    * Flushes the message queue to the server immediately if a flush is not already in progress.
    */
   async flush(callback: AnalyticsFlushCallback = () => {}): Promise<FlushResponse> {
```

Comment:
One thing I just thought of is that `FlushResponse` is no longer an accurate description of what happened during the flush. Consider:
- batch 1 flushes and encounters an error, messages are drops
- batch 2 flushes and succeeds

this would return a `FlushResponse` of `{error: undefined, batch: batch 1 messages + batch 2 messages }`

If we converted this to something like a `FlushResponse[]` we would get:
 `[{error: error1, batch: batch 1}, {error: undefined, batch: batch 2}]`
 
How does this sound for you?

Reply:
The array sounds better since it doesn't mask errors.

I also think it is worth taking a step back and deciding if we ever intend to use the return value at all.

Reply:
I don't really see a scenario where we would change what we're doing at the call site due to the presence of errors. I think it's helpful for transparency in case we need to debug something though.

Reply:
after some thought, I'm going to remove this to try and keep things simple. we can always add it back in if need be.

Reply:
I removed it due to its only function, AFAICT, being testing. But, I'm happy to convert this into a signature of `Promise<FlushResponse>` if this doesn't look great.